### DISCUSSION TITLE
Import vite library with Pinia into Nuxt project throws

### DISCUSSION THREAD

**Discussion by andredewaard** [vuejs/pinia#2216] [Help and Questions]
I have a personal library that handles authentication. 
It will be used it multiple projects. (Vite/Nuxt).

The library handles the state with Pinia and is build with Vite where Pinia is set as an external dependency
Vite.config
```ts
  build: {
    target: 'esnext',
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'index',
      fileName: 'index'
    },
    rollupOptions: {
      external: ['vue', 'pinia'],
    }
  }
```
Now when i want to use this library in my Nuxt project it works fine during development. When building for production however it throws an error.
```
[unhandled] [500] Cannot find module '{nuxt_project}/.output/server/node_modules/pinia/dist/pinia.prod.cjs' imported from {nuxt_project}/.output/server/node_modules/{library}/dist/index.js
```

Installing `@pinia/nuxt` and/or `pinia` in the Nuxt project doesn't seem to solve this issue. 

PS: The library it a private library so is not accessible for others.

**Comment by kimbaudi** [comment]
@andredewaard - I am facing the exact same issue. Have you looked at https://github.com/nuxt/nuxt/issues/19847#issuecomment-1479848806? It suggests adding the following config to `nuxt.config.ts`:

```shell
export default defineNuxtConfig({
  modules: ['@pinia/nuxt'],
  nitro: {
    alias: {
      pinia: 'pinia'
    }
  }
});
```

This seems to resolve the issue locally on build (`npm run build` or `yarn build`). However, the error still occurs for me when I try to deploy to vercel.

![image](https://github.com/nuxt/nuxt/assets/1864092/26014ff5-7b7e-4c87-b81a-2a4aeb060114)

I created a minimal repo:
- public npm package: https://github.com/kimbaudi/vue3tslibtest & https://www.npmjs.com/package/@kimbaudi/vue3tslibtest
- public github repo: https://github.com/kimbaudi/vue3-nuxt-demo

I also opened issues:
- https://github.com/nuxt/nuxt/issues/21854
- https://github.com/vuejs/pinia/issues/2284

**Comment by DylanArens** [comment]
We have the exact same issue.
We have a private library in Vue3 (with Pinia) which we want to use in multiple Nuxt3 projects. Our Nuxt3 projects have their own Pinia (@pinia/Nuxt) and Pinia in the Vue3 library is set as an external dependency in the Vite configuration (just like @andredewaard did).
Running our Nuxt project (with the library) in development modus works just fine. Building Nuxt works just fine, but when we start Nuxt (with `node .output/server/index.mjs`) it causes the same issue to show up where it can't find **pinia.prod.cjs**.
Note that the given filename is not in the location (`.output/server/node_modules/pinia/dist`), but only a file with the name "pinia"  instead of "pinia.prod". Does anyone know a workaround? Thank you in advance!

**Comment by CodeDredd** [comment]
@kimbaudi @DylanArens I don't know if it helps solving your problem, but i had this issue also in my library.
Here is the workaround i found:
https://github.com/nuxt/nuxt/issues/20128

**Reply by DylanArens** [reply]
Hmm interesting. Thank you so much! Adding Pinia with the library in the transpile array seems to work.

**Reply by CodeDredd** [reply]
You need to add all libraries to transpile which are importing pinia. Also private ones

**Reply by kimbaudi** [reply]
@CodeDredd - adding my library to build.transpile resolve the issue. Thank you! 

```shell
  build: {
    transpile: ["@kimbaudi/vue3tslibtest", "pinia"],
  },

```
